genie-core/src/main/java/com/netflix/genie/core/util/UnixProcessChecker.java

```diff
@@ -80,4 +92,32 @@ public void checkProcess() throws GenieTimeoutException, ExecuteException, IOExc
             );
         }
     }
+
+    /**
+     * Factory for {@link ProcessChecker} for UNIX systems.
+     */
+    public static class Factory implements ProcessChecker.Factory {
+
+        private final Executor executor;
+        private final boolean checkWithSudo;
+
+        /**
+         * Constructor.
+         *
+         * @param executor      The executor used by process checkers
+         * @param checkWithSudo Whether the check requires sudo
+         */
+        public Factory(final Executor executor, final boolean checkWithSudo) {
+            this.executor = executor;
+            this.checkWithSudo = checkWithSudo;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public ProcessChecker get(final int pid, final Date timeout) {
+            return new UnixProcessChecker(pid, this.executor, timeout, this.checkWithSudo);
```

Comment:
Shouldn't your factory have a check here if the OS is Unix as this process checker is dependent on that? Throw exception if on some other OS?

Reply:
When this factory is asked to produce a process checker, it does.
Checking wether this is the right environment to produce such an instance seems like overreach to me.

The checks do exist in other places:
 - In configuration, we only create this factory if the OS is unix
 - The process checker itself refuses to run unless the os is unix